spec/abci/abci++_methods.md

```diff
@@ -441,9 +440,9 @@ the consensus algorithm will use it as proposal and will not call `RequestPrepar
         * However, any resulting state changes must be kept as _candidate state_,
           and the Application should be ready to discard it in case another block is decided.
     * `RequestProcessProposal` is also called at the proposer of a round. The reason for this is to
-      inform the Application of the block header's hash, which cannot be done at `PrepareProposal`
-      time. In this case, the call to `RequestProcessProposal` occurs right after the call to
-      `RequestPrepareProposal`.
+      inform the Application of the block header's hash, which is only calculated by CometBFT after the execution of `PrepareProposal`.
+      The call to `RequestProcessProposal` occurs right after the call to `RequestPrepareProposal`.
+      However, there is no guarantee that the value of `RequestProcessProposal.txs` will be the same returned by the preceding `RequestPrepareProposal`.
```

Comment:
Considering the second bullet in the description, the following may be a better explanation.

```suggestion
    * `RequestProcessProposal` is also called at the proposer of a round. The reason for this is to
      inform the Application of the block header's hash, which is only calculated by CometBFT after the execution of `PrepareProposal`.
      Normally the call to `RequestProcessProposal` occurs right after the call to `RequestPrepareProposal` and
      `RequestProcessProposal` matches the block produced based on `ResponsePrepareProposal` (i.e.,
      `RequestPrepareProposal` equals `RequestProcessProposal.txs`).
      However, no such guarantee is made since, in the presence of failures, `RequestProcessProposal` may match
      `ResponsePrepareProposal` from an earlier invocation or `ProcessProposal` may not be invoked at all.
```

Reply:
I would remove the sentence "the reason for this is to inform the Application of the block header's hash, which is only calculated by CometBFT after the execution of `PrepareProposal`."

Given the current situation, it only adds confusion IMO